### PR TITLE
(WIP) Add FL Studio

### PR DESCRIPTION
**Pull request by BobyMCbobs**
[Image-Line.com - FL Studio](https://image-line.com/flstudio)

TODO:
- [ ] Get 32-bit Wine / Compat32 to work, so Wine can run both 32-bit and 64-bit apps
- [ ] Contact Image-Line to get their thoughts on packaging this

Note: this package doesn't contain an installed version, but is a wrapper for running the installer and launching FL Studio.

I would much appreciate some help with the first stated to do.

### CODE REVIEW

File: com.imageline.flstudio.appdata.xml [hunk 1]
```diff
@@ -0,0 +1,69 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<component type="desktop">
+    <id>com.imageline.flstudio.desktop</id>
```
Comment (nedrichards) [inline com.imageline.flstudio.appdata.xml]: have a look at the id guidelines on https://github.com/flathub/flathub/wiki/AppData-Guidelines the .desktop bit isn't recommended (you'll need to add another bit to say that, but it'll be better overall)

**Comment by nedrichards** [comment]
bot, build com.imageline.flstudio

File: com.imageline.flstudio.yml [hunk 2]
```diff
@@ -0,0 +1,160 @@
+app-id: com.imageline.flstudio
+branch: stable
+
+runtime: org.freedesktop.Platform
+runtime-version: 18.08
```
Comment (LeandroStanger) [inline com.imageline.flstudio.yml]: Update runtime to 19.08

File: com.imageline.flstudio.yml [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+      - install -Dm644 com.imageline.flstudio.desktop /app/share/applications/com.imageline.flstudio.desktop
+      - install -Dm644 com.imageline.flstudio.256x256.png /app/share/icons/hicolor/256x256/apps/com.imageline.flstudio.png
+      - install -Dm644 com.imageline.flstudio.128x128.png /app/share/icons/hicolor/128x128/apps/com.imageline.flstudio.png
+      - install -Dm644 com.imageline.flstudio.64x64.png /app/share/icons/hicolor/64x64/apps/com.imageline.flstudio.png
+      - install -Dm644 com.imageline.flstudio.48x48.png /app/share/icons/hicolor/48x48/apps/com.imageline.flstudio.png
+      - install -Dm644 com.imageline.flstudio.32x32.png /app/share/icons/hicolor/32x32/apps/com.imageline.flstudio.png
+      - install -Dm644 com.imageline.flstudio.16x16.png /app/share/icons/hicolor/16x16/apps/com.imageline.flstudio.png
+    sources:
+      - type: file
+        path: com.imageline.flstudio.appdata.xml
```
Comment (LeandroStanger) [inline com.imageline.flstudio.yml:141]: New appdata files should use metainfo instead. https://www.freedesktop.org/software/appstream/docs/chap-Quickstart.html

**Comment by LeandroStanger** [comment]
bot, build com.imageline.flstudio

File: com.imageline.flstudio.yml (hunk 2 above)
Comment (hfiguiere) [reply com.imageline.flstudio.yml:5]: 20.08 actually.

**Comment by BobyMCbobs** [comment]
Still not sure how to solve this yet.
Would like to revisit again at a later date.

/open

**Pull request closed without merging**